ext/scheduler/airflow2/airflow.go

```diff
@@ -213,24 +220,21 @@ func (s *scheduler) GetJobStatus(ctx context.Context, projSpec models.ProjectSpe
 	}
 
 	fetchURL := fmt.Sprintf(fmt.Sprintf("%s/%s", schdHost, dagStatusUrl), jobName)
-	request, err := http.NewRequestWithContext(ctx, http.MethodGet, fetchURL, nil)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to build http request for %s", fetchURL)
+	req := airflowRequest{
+		URL:    fetchURL,
+		method: http.MethodGet,
+		token:  authToken,
+		body:   nil,
 	}
-	request.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(authToken))))
-
-	resp, err := s.httpClient.Do(request)
+	resp, err := s.callAirflow(ctx, req)
```

Comment:
Can you move the call airflow method on the airflowRequest struct, that would be the right place.

Reply:
Done and renamed the method to `invoke`

Reply:
how about initializing the request with the http client itself or creating the client inside as part of invoke call?

Reply:
as discussed, moved the client code to separate file